tuf/tuf_test.go

```diff
@@ -941,15 +864,215 @@ func TestAddBaseKeysToRoot(t *testing.T) {
+
+	_, err = repo.GetBaseRole("targets/delegation")
+	assert.Error(t, err)
+}
+
+func TestGetDelegationValidRoles(t *testing.T) {
+	ed25519 := signed.NewEd25519()
+	repo := initRepo(t, ed25519)
+
+	testKey1, err := ed25519.Create("targets/test", data.ED25519Key)
+	assert.NoError(t, err)
+	role, err := data.NewRole("targets/test", 1, []string{testKey1.ID()}, []string{"path", "anotherpath"})
+	assert.NoError(t, err)
+
+	err = repo.UpdateDelegations(role, data.KeyList{testKey1})
+	assert.NoError(t, err)
+
+	delgRole, err := repo.GetDelegationRole("targets/test")
+	assert.NoError(t, err)
+	assert.Equal(t, "targets/test", delgRole.Name)
+	assert.Equal(t, 1, delgRole.Threshold)
+	assert.Equal(t, []string{testKey1.ID()}, delgRole.ListKeyIDs())
+	assert.Equal(t, []string{"path", "anotherpath"}, delgRole.Paths)
+	assert.Equal(t, testKey1, delgRole.Keys[testKey1.ID()])
+
+	testKey2, err := ed25519.Create("targets/a", data.ED25519Key)
+	assert.NoError(t, err)
+	role, err = data.NewRole("targets/a", 1, []string{testKey2.ID()}, []string{""})
+	assert.NoError(t, err)
+
+	err = repo.UpdateDelegations(role, data.KeyList{testKey2})
+	assert.NoError(t, err)
 
-	// Clear the keysDB, check that we get an empty list
-	repo.keysDB = keys.NewDB()
-	roles = repo.GetAllLoadedRoles()
-	assert.Len(t, roles, 0)
+	delgRole, err = repo.GetDelegationRole("targets/a")
+	assert.NoError(t, err)
+	assert.Equal(t, "targets/a", delgRole.Name)
+	assert.Equal(t, 1, delgRole.Threshold)
+	assert.Equal(t, []string{testKey2.ID()}, delgRole.ListKeyIDs())
+	assert.Equal(t, []string{""}, delgRole.Paths)
+	assert.Equal(t, testKey2, delgRole.Keys[testKey2.ID()])
+
+	testKey3, err := ed25519.Create("targets/test/b", data.ED25519Key)
+	assert.NoError(t, err)
+	role, err = data.NewRole("targets/test/b", 1, []string{testKey3.ID()}, []string{"path/subpath", "anotherpath"})
+	assert.NoError(t, err)
+
+	err = repo.UpdateDelegations(role, data.KeyList{testKey3})
+	assert.NoError(t, err)
+
+	delgRole, err = repo.GetDelegationRole("targets/test/b")
+	assert.NoError(t, err)
+	assert.Equal(t, "targets/test/b", delgRole.Name)
+	assert.Equal(t, 1, delgRole.Threshold)
+	assert.Equal(t, []string{testKey3.ID()}, delgRole.ListKeyIDs())
+	assert.Equal(t, []string{"path/subpath", "anotherpath"}, delgRole.Paths)
+	assert.Equal(t, testKey3, delgRole.Keys[testKey3.ID()])
+
+	testKey4, err := ed25519.Create("targets/test/c", data.ED25519Key)
+	assert.NoError(t, err)
+	// Try adding empty paths, ensure this is valid
+	role, err = data.NewRole("targets/test/c", 1, []string{testKey4.ID()}, []string{})
+	assert.NoError(t, err)
+
+	err = repo.UpdateDelegations(role, data.KeyList{testKey3})
+	assert.NoError(t, err)
+}
+
+func TestGetDelegationRolesInvalidName(t *testing.T) {
+	ed25519 := signed.NewEd25519()
+	repo := initRepo(t, ed25519)
+
+	_, err := repo.GetDelegationRole("invalid")
+	assert.Error(t, err)
+
+	for _, role := range data.BaseRoles {
+		_, err = repo.GetDelegationRole(role)
+		assert.Error(t, err)
+	}
```

Comment:
Can we also add a check for "targets/delegation" here?  Since no delegation role was specified, this should also return `ErrInvalidRole` since it's not in the targets tree hierarchy